econml/validate/drtester.py

```diff
@@ -0,0 +1,796 @@
+            control treatment. It is recommended to have the control take value 0 and all other treatments be integers
+            starting at 1
+        y: vector of length n
+            Outcomes
+        reg_preds: (n x n_treat) matrix
+            Outcome predictions for each potential treatment
+        prop_preds: (n x n_treat) matrix
+            Propensity score predictions for each treatment
+
+        Returns
+        -------
+        Doubly robust outcome values
+        """
+
+        # treat each treatment as a separate regression
+        # here, prop_preds should be a matrix
+        # with rows corresponding to units and columns corresponding to treatment statuses
+        dr_vec = []
+        d0_mask = np.where(D == 0, 1, 0)
+        y_dr_0 = reg_preds[:, 0] + (d0_mask / np.clip(prop_preds[:, 0], .01, np.inf)) * (y - reg_preds[:, 0])
+        for k in np.sort(np.unique(D)):  # pick a treatment status
+            if k > 0:  # make sure it is not control
+                dk_mask = np.where(D == k, 1, 0)
+                y_dr_k = reg_preds[:, k] + (dk_mask / np.clip(prop_preds[:, k], .01, np.inf)) * (y - reg_preds[:, k])
+                dr_k = y_dr_k - y_dr_0  # this is an n x 1 vector
+                dr_vec.append(dr_k)
+        dr = np.column_stack(dr_vec)  # this is an n x n_treatment matrix
+
+        return dr
+
+        # Fits CATE in training, predicts in validation
+    def fit_cate_train(
+        self,
+        reg_cate,
+        Ztrain: np.array,
+        Zval: np.array
+    ) -> np.array:
+
+        """
+        Fit provided CATE estimator on training dataset and return predicted out-of-sample CATE values for validation
+        set.
+
+        Parameters
+        ----------
+        reg_cate: estimator
+            CATE model. Must be able to implement `fit' and `predict' methods
+        Ztrain: (n_train x k) matrix
+            Training sample features for CATE model
+        Zval: (n_val x k) matrix
+            Validation sample features for CATE model
+
+        Returns
+        -------
+        n_val x n_treatment array of predicted CATE values for validation set.
+        """
+
+        if np.ndim(Zval) == 1:
+            Zval = Zval.reshape(-1, 1)
+
+        if np.ndim(Ztrain) == 1:
+            Ztrain = Ztrain.reshape(-1, 1)
+
+        if self.n_treat == 1:
+            reg_cate_fitted = reg_cate.fit(Ztrain, np.squeeze(self.dr_train))
+            cate_preds = np.expand_dims(reg_cate_fitted.predict(Zval), axis=1)
+        else:
+            cate_preds = []
+            for k in range(self.n_treat):  # fit a separate cate model for each treatment status?
+                reg_cate_fitted = reg_cate.fit(Ztrain, self.dr_train[:, k])
+                cate_preds.append(reg_cate_fitted.predict(Zval))
+
+            cate_preds = np.column_stack(cate_preds)
+
+        return cate_preds
+
+    # CV prediction of CATEs
+    @staticmethod
+    def fit_cate_cv(
```

Comment:
I suspect all these methods related to CATE fitting should not be here. you should only be accepting fitted cate models.